models/management.cattle.io.cluster.js

```diff
@@ -236,14 +236,16 @@ export default class MgmtCluster extends HybridModel {
       return undefined;
     }
 
-    const color = this.metadata?.annotations[CLUSTER_BADGE.COLOR] || '#7f7f7f';
-    const iconText = this.metadata?.annotations[CLUSTER_BADGE.ICON_TEXT] || '';
+    const color = this.metadata?.annotations[CLUSTER_BADGE.COLOR];
+    const useForIcon = this.metadata?.annotations[CLUSTER_BADGE.USE_AS_ICON] || '';
+    const letter = this.metadata?.annotations[CLUSTER_BADGE.ICON_TEXT] || text.substr(0, 1).toUpperCase();
```

Comment:
This only take first char - we support up to 2 - which is what the `iconText` was doing before (see line 246)